plugins/csp-wms-overlays/src/TextureOverlayRenderer.hpp

```diff
@@ -0,0 +1,167 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2020 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_OVERLAYS_TEXTURE_OVERLAY_RENDERER_HPP
+#define CSP_WMS_OVERLAYS_TEXTURE_OVERLAY_RENDERER_HPP
+
+#include "Plugin.hpp"
+#include "WebMapLayer.hpp"
+#include "WebMapService.hpp"
+#include "WebMapTextureLoader.hpp"
+
+#include <VistaKernel/GraphicsManager/VistaOpenGLDraw.h>
+#include <VistaMath/VistaBoundingBox.h>
+#include <VistaOGLExt/VistaGLSLShader.h>
+
+#include <array>
+#include <functional>
+#include <future>
+#include <unordered_map>
+
+// FORWARD DEFINITIONS
+class VistaGLSLShader;
+class VistaOpenGLNode;
+class VistaTexture;
+class VistaViewport;
+
+namespace cs::core {
+class SolarSystem;
+class TimeControl;
+class Settings;
+} // namespace cs::core
+
+namespace csp::wmsoverlays {
+
+/// Class which gets a geo-referenced texture and overlays if onto the previous rendered scene.
+/// Therefore it copies the depth buffer first. Second, in the shader it does an inverse projection
+/// to get the cartesian coordinates. This coordinates are transformed to latitude and longitude to
+/// do the lookup in the geo-referenced texture. The value is then overlayed on that pixel position.
+class TextureOverlayRenderer : public IVistaOpenGLDraw {
+ public:
+  TextureOverlayRenderer(std::string center, std::shared_ptr<cs::core::SolarSystem> solarSystem,
+      std::shared_ptr<cs::core::TimeControl> timeControl,
+      std::shared_ptr<cs::core::Settings>    settings,
+      std::shared_ptr<Plugin::Settings>      pluginSettings);
+  virtual ~TextureOverlayRenderer();
```

Comment:
Use either override or final for inherited virtual functions.
```suggestion
  ~TextureOverlayRenderer() override;
```
```suggestion
  ~TextureOverlayRenderer() final;
```